entropy.go

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot
+
+import (
+	gpv "github.com/canonical/go-password-validator"
+)
+
+type PassphraseEntropyStats struct {
+	SymbolPoolSize  uint32
+	NumberOfSymbols uint32
+	EntropyBits     uint32
+}
+
+// CheckPassphraseEntropy calculates entropy for PINs and passphrases (PINs will be supplied as a numeric passphrase).
+func CheckPassphraseEntropy(passphrase string) (*PassphraseEntropyStats, error) {
+	stats := &PassphraseEntropyStats{}
```

Comment:
It doesn't matter here and tbh I do this inconsistently as well so this is more of a note than a request for any change, but you can just do `new(PassphraseEntropyStats)` here. You can even just declare it as a stack variable - `var stats PassphraseEntropyStats` - and the toolchain recognizes that it escapes the current context and stack allocates it automatically.